doc/source/whatsnew/v0.21.1.txt

```diff
@@ -7,6 +7,45 @@ This is a minor release from 0.21.1 and includes a number of deprecations, new
 features, enhancements, and performance improvements along with a large number
 of bug fixes. We recommend that all users upgrade to this version.
 
+.. _whatsnew_0211.special:
+
+Restore Matplotlib datetime Converter Registration
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Pandas implements some matplotlib converters for nicely formatting the axis
+labels on plots with ``datetime`` or ``Period`` values. Prior to pandas 0.21.0,
+these were implicitly registered with matplotlib, as a side effect of
+``import pandas``. In pandas 0.21.0, we required users to explicitly register
+the converter.
+
+.. code-block:: python
+
+   >>> from pandas.tseries import converter
+   >>> converter.register()
+
+This caused problems for some users, so we're temporarily reverting that change;
+pandas will again register the converters on import. Using the converters
+without explicitly registering the formatters will cause a ``FutureWarning``:
+
+.. code-block:: python
+
+   >>> import pandas as pd
+   >>> import matplotlib.pyplot as plt
+   >>> fig, ax = plt.subplots()
+   >>> ax.plot(pd.Series(range(12), index=pd.date_range('2017', periods=12)))
+   FutureWarning: Using an implicitly registered datetime converter for a
+   matplotlib plotting method. The converter was registered by pandas on import.
+   Future versions of pandas will require you to explicitly register matplotlib
+   converters.
+ 
+   To register the converters:
+           >>> from pandas.tseries import converter
+           >>> converter.register()
+
+As the error message says, you'll need to register the converts if you intend to
+use them with matplotlib plotting functions. Pandas plotting functions, such as
+``Series.plot``, will register them for you. (:issue:`18301`)
```

Comment:
maybe a bit more explicit like: "When using pandas plotting functions, such as Series.plot, the converters will be registered for you and you do not need to do this explicitly"